### ISSUE TITLE
Question: interpreting association results

### ISSUE THREAD

**Issue by PhilPalmer** [h3abionet/h3agwas#14]
Hi, I am trying to run the association pipeline with the test data included in the `sample` directory.

I am familiar with nextflow and docker but less familiar with GWAS/plink etc. 

I have run the following command:
```
nextflow run plink-assoc.nf --input_dir sample --input_pat sampleA --chi2 1 --logistic 1 --adjust 1 --data sample/sample.phe --pheno "SEX,PHE" --covariates "PAT,MAT" -profile docker
```

It ran to completion producing a very nice report: 
[out-report.pdf](https://github.com/h3abionet/h3agwas/files/2783786/out-report.pdf)

However, I am not completley sure I understand exactly what the pipeline is doing.

For example how are the `pheno` & `covariates` parameters are used. 
The documentation says the following:
>pheno: a comma-separated list of phenotypes that you want to test. Each phenotype is tested separately.
covariates: a comma-separated list of phenotypes that you want to use

So are each of the `pheno` paramters tested against the `covariates` parameters to see if the vary together at the particular genetic regions?

Also what are the observed and expected p-value in the output report for? Is this for each trait/phenotype that is used in the covariants and how strongly they correlate with particular genomic regions for example? And why are some of the plots empty?

Great pipeline though and many thanks in advance. Any help would be much appreciated

**Comment by shaze** [comment]
Apologies for the slow response. I was travelling and them issed the email on my return.

Each phentoype is tested against the genotypes with the same covariates

So if you said `--pheno BMI,height --covariates=sex,age` then you would run a GWAS on the data to find BMI associations using sex and age as covariates; AND you would run a GWAS on the data to find height associations using sex and age as covariates.

It would not make sense to have  the same attribute in both sets.  If you need to use different covariates for different charactertistics (say smoking for BMI but not height) then you would have to do this in separate runs.

In summary: you are "tesing" association with the named phenotype "using" the named covariates.

Sorry again for the delay -- let me know if anything is still unclear

**Comment by PhilPalmer** [comment]
Hi @shaze, 

Okay, that's great thanks for your response.

Do you happen to know why the plots produced for the logistic regression are empty? (As shown by the report in my question above)

Is it a problem a problem with my input parameters?

Thanks again

**Comment by shaze** [comment]
Dear Phil

The underlying problem is that is doesn't make sense to use MAT and PAT as the covariates and so there are not valid statistical results. Try using SEX as a covariant.

Although the pipeline does try to give meaningful feedback, it is important not to treat it completely as a black box.

For example, when I ran it I got something like this
```
nextflow run plink-assoc.nf --input_dir sample --input_pat sampleA --chi2 1 --logistic  1 --adjust 1 --data sample/sample.phe --pheno SEX,PHE --covariates PAT,MAT
N E X T F L O W  ~  version 18.10.1
Launching `plink-assoc.nf` [shrivelled_fourier] - revision: af7ae94175

Testing data            : sampleA

Testing for phenotypes  : SEX,PHE

Using covariates        : PAT,MAT


Doing logistic regression testing


[warm up] executor > local
[b4/3d6a60] Submitted process > extractPheno (1)
[93/839695] Submitted process > computePCA (1)
[bb/c0dd9d] Submitted process > showPhenoDistrib (1)
[35/15ccd1] Submitted process > computeTest (2)
[b5/898d65] Submitted process > computeTest (1)
Completed plink test logistic
[3e/7858fc] Submitted process > drawPlinkResults (1)
Completed plink test logistic
[64/a5f2b1] Submitted process > drawPlinkResults (2)
[e0/285e9a] Submitted process > drawPCA (1)
[9f/2924fd] Submitted process > doReport
```

You are shown each step and the directory where the work is done. If you looked at `work/b5/898d65..` you would see the work directory for one of the logistic tests. You could then see the output of that test and you would very quickly see the problem.

The issue you have raised does highlight a weakness in the reporting and better feedback should be given in this case.

Thanks for raising this

**Comment by PhilPalmer** [comment]
Ah, that makes sense now.

Thank you very much for your help
